### PR TITLE
KAFKA-8807: Flaky GlobalStreamThread test

### PR DESCRIPTION
**Pull request by bbejeck**
A minor refactor to explicitly verify that `Processor#close` is only called once.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: streams/src/test/java/org/apache/kafka/streams/integration/GlobalThreadShutDownOrderTest.java [hunk 1]
```diff
@@ -49,14 +49,26 @@
...
 import java.util.List;
 import java.util.Properties;
+import java.util.concurrent.atomic.AtomicInteger;
 
 import static org.junit.Assert.assertEquals;
 
+/**
+ * This test asserts that when Kafka Streams is closing and shuts
+ * down a StreamThread the closing of the GlobalStreamThread happens
+ * after all the StreamThreads are completely stopped.
```
Comment (bbejeck) [inline streams/src/test/java/org/apache/kafka/streams/integration/GlobalThreadShutDownOrderTest.java:59]: Added a brief description as when I revisited the test I needed to look at the code to remember what the test was for.

File: streams/src/test/java/org/apache/kafka/streams/integration/GlobalThreadShutDownOrderTest.java [hunk 2]
```diff
@@ -188,6 +201,7 @@ public void process(final String key, final Long value) {
 
         @Override
         public void close() {
+            closeCounter.getAndIncrement();
```
Comment (bbejeck) [inline streams/src/test/java/org/apache/kafka/streams/integration/GlobalThreadShutDownOrderTest.java:204]: This should only ever get incremented once

**Comment by bbejeck** [comment]
ping @guozhangwang, @ableegoldman, and @cadonna 

I ran this test locally 100x on a couple of occasions and I can't reproduce the error.  I'll follow up with a system test to see if there is an issue.   But IMHO the system test is separate and we can merge this PR now.

**Review by cadonna — COMMENTED** [review]
I think it is worthwhile to try to reformulate this test as a unit test in `KafkaStreamsTest`. It should be possible to check the order of the `shutdown()` and `join()` calls by using `EasyMock.createStrictControl()` as described here http://easymock.org/user-guide.html#verification-order, but I have never tried it.

**Review by ableegoldman — APPROVED** [review]
Thanks Bill! This is a good improvement to be able to pick out actual bugs here going forward. LGTM

**Review by guozhangwang — APPROVED** [review]

**Comment by guozhangwang** [comment]
LGTM, thanks @bbejeck . Please feel free to merge.

**Pull request merged**